link-grammar/parse/prune.c

```diff
@@ -904,6 +904,25 @@ static bool check_null_word(prune_context *pc, int w)
 	return false;
 }
 
+/**
+ * Return \c true if a pass has not made any changes.
+ * Also print pruning pass stats.
+ */
+static bool pruning_pass_end(prune_context *pc, int *prune_total)
+{
+	int total = pc->N_deleted[0] + pc->N_deleted[1];
+
+	lgdebug(D_PRUNE, "Debug: r->l pass changed %d and deleted %d (%d+%d)\n",
+	        pc->N_changed, total, pc->N_deleted[0], pc->N_deleted[1]);
+	if (pc->ml != NULL)
+		lgdebug(D_PRUNE, "Debug: xlink: %d\n", pc->N_xlink);
+
+	pc->N_changed = pc->N_deleted[0] = pc->N_deleted[1] = pc->N_xlink = 0;
+
+	*prune_total += total;
+	return ((pc->N_changed == 0) && (total == 0));
```

Comment:
you  set `pc->N_changed` to zero up above.  The old code did this test before setting it to zero.